### ISSUE TITLE
Grabber crashes during second run

### ISSUE THREAD

**Issue by theHubi** [Bionus/imgbrd-grabber#395]
At first I thought the new version fixed all the crashes. Apparently I was wrong.
I should've sent you about 10 automated bug reports which come from the following:

30 Downloads in the queue -> i select those with under 1000 images and click "Download selected".
Download finishes like usual with popup. -> I now click on "Download" to download the other 4000 images left in the queue.
Without even fully loading all pages, the Grabber crashes.

So to sum up: First run, everything works, second run, Grabber crashes.

**Comment by theHubi** [comment]
I just tested some more:
Apparently the "Download selected"-Button does not work twice. Meaning if you download with the "Selected"-Button and then close the batch-downloader, select another source and click again, nothing happens. And if you click "Download"(All), it crashes as well.

**Comment by Bionus** [comment]
> Without even fully loading all pages, the Grabber crashes.

While trying to reproduce your issue I indeed ended up crashing, but not where you said it would.
It crashed a

**Comment by theHubi** [comment]
It seems that this issue appears randomly. Not every run but every couple runs. Maybe a loop just gets stuck. I don't know.
But I can confirm your statement. Yesterday it crashed about 10 times before downloading the last image. Out of those 10 times, 5 times the "Download Selected" button did not open the batch grabber. The Download (All)-Button however did.

**Comment by theHubi** [comment]
Hey dude. I think I figured it out. I tested around a bit. After downloading once it won't start doing so a second time. But it starts the process. Grabber cannot re-allocate either CPU or RAM after the download. I just "started" the download of 8000 files 3 times.. without doing anything it went up all the way to 2 GB of RAM usage but didn't utilize any CPU.
So somewhere once Grabber gives back the CPU it doesn't ask for it again.

**Comment by ItsSohma** [comment]
@theHubi No longer valid? I know @Bionus has made several changes to the download process.

**Comment by Bionus** [comment]
Yes indeed, there has been huge changes since July 2015 related to RAM usage. First by reducing the memory footprint of the program, then by automatically making smaller download batches, and finally in the last update by not even loading downloaded images in RAM.

If the issue persists in the last version, feel free to comment here or open another issue.

_PS: thanks @ItsSohma 👍_
